tasks/miarecweb.yml

```diff
@@ -14,38 +14,52 @@
     path: "{{ deploy_helper.new_release_path }}/app"
   register: miarecweb_app_dir_exists
 
-
-- name: Add an unfinished file, to allow cleanup on successful finalize
-  file:
-    path: '{{ deploy_helper.new_release_path }}/{{ deploy_helper.unfinished_filename }}'
-    state: touch
-    mode: '0644'
-  when: not miarecweb_app_dir_exists.stat.exists
-
-
-- name: Download miarecweb tarball
-  get_url:
-    url: "{{ miarecweb_download_base_url }}/{{ miarecweb_tarball_file | replace('+', '%2b') }}"
-    dest: "{{ miarecweb_download_dir }}/{{ miarecweb_tarball_file }}"
-    mode: '0644'
-  when: not miarecweb_app_dir_exists.stat.exists
-
-
-- name: Untar the source package
-  unarchive:
-    src: "{{ miarecweb_download_dir }}/{{ miarecweb_tarball_file }}"
-    dest: "{{ miarecweb_download_dir }}/"
-    creates: "{{ miarecweb_download_dir }}/miarecweb-{{ miarecweb_version }}"
-    copy: false
-    mode: u=rwX,g=rX,o=rX
-  when: not miarecweb_app_dir_exists.stat.exists
-
-
-- name: Move app directory to releases folder
-  command: mv "{{ miarecweb_download_dir }}/miarecweb-{{ miarecweb_version }}" "{{ deploy_helper.new_release_path }}/app"
-  args:
-    creates: "{{ deploy_helper.new_release_path }}/app"
+- name: Download and extract source package.
   when: not miarecweb_app_dir_exists.stat.exists
+  block:
+
+    - name: Add an unfinished file, to allow cleanup on successful finalize
+      file:
+        path: '{{ deploy_helper.new_release_path }}/{{ deploy_helper.unfinished_filename }}'
+        state: touch
+        mode: '0644'
+      when: not miarecweb_app_dir_exists.stat.exists
+
+
+    - name: Download miarecweb tarball
+      get_url:
+        url: "{{ miarecweb_download_base_url }}/{{ miarecweb_tarball_file | replace('+', '%2b') }}"
+        dest: "{{ miarecweb_download_dir }}/{{ miarecweb_tarball_file }}"
+        mode: '0644'
+      when: not miarecweb_app_dir_exists.stat.exists
+
+    # unarchive must change the permissions of the destination folder when mode parameter is supplied.
+    # But it behaves differently with tar.gz and zip archives.
+    # When extracting tar.gz, it doesn't touch the destination folder's permissions.
+    # While in case of extracting zip into something like /tmp/ with mode: 755, it overrides the permissions of /tmp to 755.
+    # This step will create a direcoty specific to this role to extract files to
+    - name: Create directory for extracted files
+      file:
+        path: "{{ miarecweb_download_dir }}/miarec-{{ miarecweb_version }}"

```

Comment:
It should be "miarecweb" not "miarec" to avoid potential conflicts when "miarec" recorder project is deployed with the same version on the same server

Reply:
I see one issue here.
For each software update, we create a subfolder "{{ miarecweb_download_dir }}/miarec-{{ miarecweb_version }}", which is never deleted.

It is better to create the same folder (unique for project): 

```
    - name: Create directory for extracted files
      file:
        path: "{{ miarecweb_download_dir }}/miarecweb"
```